modules/migration-migrating-applications-api.adoc

```diff
@@ -0,0 +1,413 @@
+
+* Indirect migration: Images, volumes, and Kubernetes objects are copied from the `source` cluster to the replication repository and then from the replication repository to the `destination` cluster.
+* Direct migration: Images or volumes are copied directly from the `source` cluster to the `destination` cluster. Direct image migration and direct volume migration have significant performance benefits.
+
+You create the following custom resources (CRs) to perform a migration:
+
+* `MigCluster`: Defines a `host`, `local`, or `remote` cluster
+* `Secret`: Contains credentials for a `remote` cluster or storage
+* `MigStorage`: Defines a replication repository
+* `MigPlan`: Defines a migration plan
+* `MigMigration`: Performs a migration defined in an associated `MigPlan`
++
+You can create multiple `MigMigration` CRs for a single `MigPlan` CR for the following purposes:
++
+* To perform stage migrations, without quiescing the application, before running a full migration. Stage migrations copy most of the data to the `destination` cluster, which improves the performance of the full migration.
+* To cancel a migration in progress
+* To roll back a completed migration
+
+.Prerequisites
+
+* You must have `cluster-admin` privileges for all clusters.
+* You must install the {product-full} CLI (`oc`).
+* You must have an object storage to configure as a replication repository.
+* If you are using direct image migration, you must expose a secure registry route on all remote clusters.
+* You must install the same version of the {mtc-full} Operator on all clusters.
+
+.Procedure
+
+. Create a `MigrationController` CR manifest for the `host` cluster called `host-migration-controller.yaml`:
++
+[source,yaml]
+----
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigrationController
+metadata:
+  name: migration-controller
+  namespace: openshift-migration
+spec:
+  azure_resource_group: <azure_resource_group> <1>
+  cluster_name: host
+  migration_velero: true
+  migration_controller: true
+  migration_ui: true
+  migration_log_reader: true
+  restic_timeout: 1h
+
+  mig_pv_limit: 100
+  mig_pod_limit: 100
+  mig_namespace_limit: 10
+----
+<1> Specify the Azure resource group if you are using Microsoft Azure for storage.
+
+. Create a `MigrationController` CR for the `host` cluster:
++
+[source,terminal]
+----
+$ oc create -f host-migration-controller.yaml -n openshift-migration
+----
+
+. Create a `MigCluster` CR manifest for the `host` cluster called `host-cluster.yaml`:
++
+[source,yaml]
+----
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigCluster
+metadata:
+  name: host
+  namespace: openshift-migration
+spec:
+  isHostCluster: true
+----
+
+. Create the `MigCluster` CR for the `host` cluster:
++
+[source,terminal]
+----
+$ oc create -f host-cluster.yaml -n openshift-migration
+----
+
+. Create a `MigrationController` CR manifest for each remote cluster called `remote-migration-controller.yaml`:
++
+[source,yaml]
+----
```

Comment:
The full MigrationController resource is like this

https://github.com/konveyor/mig-operator/blob/master/deploy/non-olm/controller-3.yml

How to create the right MigrationController CRs is described in the installation part of the documentation. Probably we should send the user there. https://deploy-preview-29058--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_3_4/deploying-cam-3-4.html